templates/service/test/controllers/MicroserviceHelloWorldControllerSpec.scala

```diff
@@ -2,14 +2,25 @@ package uk.gov.hmrc.$!APP_PACKAGE_NAME!$.controllers
 
 import org.scalatest.matchers.should.Matchers
 import org.scalatest.wordspec.AnyWordSpec
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import play.api.Application
 import play.api.http.Status
 import play.api.test.Helpers._
-import play.api.test.{FakeRequest, Helpers}
+import play.api.test.FakeRequest
+import play.api.inject.guice.GuiceApplicationBuilder
 
-class MicroserviceHelloWorldControllerSpec extends AnyWordSpec with Matchers {
+class MicroserviceHelloWorldControllerSpec extends AnyWordSpec with Matchers with GuiceOneAppPerSuite {
+
+  override def fakeApplication(): Application =
+    new GuiceApplicationBuilder()
+      .configure(
+        "metrics.jvm"     -> false,
+        "metrics.enabled" -> false
+      )
+      .build()
 
   private val fakeRequest = FakeRequest("GET", "/")
-  private val controller = new MicroserviceHelloWorldController(Helpers.stubControllerComponents())
+  private val controller = app.injector.instanceOf[MicroserviceHelloWorldController]
```

Comment:
I think loading the whole app to test one component is not a practice we want to promote - it makes the tests a lot slower, and seems to be done here in order to test other non-associated parts of the system (e.g. does the app start) and increase code coverage.
I think this test should stay as it is, `ErrorHandlerSpec` should also be updated to only create the `ErrorHandler`.
We should create a new test, which is explicit in it's intention, that it exists to test the app starts as a whole - e.g. `MicroserviceIntegSpec`

Reply:
+1

Reply:
Hi @colin-lamed , the ErrorHandler has a dependency on the error template which in turn has a dependency on the layout which has a dependency on AppConfig. Testing any controller or view purely through unit tests without injecting instances using a fake application seems unnecessarily difficult, hence the use of an application.

I totally agree with not starting up an app for tests when not needed. If this were a model, service or connector I'd typically mock things out but here the application is needed.

Reply:
WRT `MicroserviceIntegSpec`, I'll add the test and revert the changes to `MicroserviceHelloWorldController` and the associated test.